packages/xstate-wallet/src/store/memory-store.ts

```diff
@@ -285,17 +302,18 @@ export class MemoryStore implements Store {
         })
       );
     }
-
+    const dbObjectives = await this.backend.objectives();
     objectives?.forEach(objective => {
-      if (!_.includes(this._objectives, objective)) {
-        this._objectives.push(objective);
+      if (!_.includes(objectives, objective)) {
+        objectives.push(objective);
         this._eventEmitter.emit('newObjective', objective);
       }
     });
+    this.backend.setObjectives(dbObjectives);
   }
```

Comment:
I think the backend should accept a list of objectives and return the list of new objectives. This could then be
```
(await this.backend.setObjectives(objectives)).forEach(o => this._eventEmitter.emit('newObjective', o);)
```

Otherwise, this code is not safe as asynchronous code.